home/utils/AccountNameContext.tsx

```diff
@@ -0,0 +1,18 @@
+import { createContext, useContext } from 'react';
+
+export type AccountNameContextValue = {
+  accountName?: string;
+  setAccountName: (accountName?: string) => void;
+};
+
+export const AccountNameContext = createContext<AccountNameContextValue | null>(null);
+
```

Comment:
I think its pretty common to colocate a `Provider` type component as well so that you can see the logic related to the setState for this context in one place - this might clean up some of the implementation in HomeApp related to finding the right account name, for example `<AccountNameProvider isAuthenticated={isAuthenticated} currentUserQuery={...}  {...whateverElse} />` - this also makes it a bit easier to find business logic related to account name - just my opinion on it